rust/pact_mock_server_ffi/src/lib.rs

```diff
@@ -111,6 +108,56 @@ pub unsafe extern fn init(log_env_var: *const c_char) {
   builder.try_init().unwrap_or(());
 }
 
+/// Sets the log level explicitly
+///
+/// # Safety
+///
+/// Exported functions are inherently unsafe.
+#[no_mangle]
+pub unsafe extern fn log_level(level: *const c_char) {
+  let mut builder = Builder::from_default_env();
+  let log_level = log_level_from_c_char(level);
+
+  builder.filter_level(log_level.to_level_filter());
+  builder.try_init().unwrap_or(());
+}
+
+/// Log using the shared core logging facility, instead
+///
+/// * `source` - String. The source of the log, such as the class or caller framework to
+///                      disambiguate log lines from the rust logging (e.g. pact_go)
+/// * `log_level` - String. One of TRACE, DEBUG, INFO, WARN, ERROR
+/// * `message` - Message to log
+///
+/// Exported functions are inherently unsafe.
+#[no_mangle]
+pub unsafe extern fn log(source: *const c_char, log_level: *const c_char, message: *const c_char) {
+  let target = convert_cstr("target", source).unwrap_or("client");
+
+  if !message.is_null() {
+    match convert_cstr("message", message) {
+      Some(message) => log!(target: target, log_level_from_c_char(log_level), "{}", message),
+      None => ()
+    }
+  }
+}
+
+unsafe fn log_level_from_c_char(log_level: *const c_char) -> log::Level {
+  if !log_level.is_null() {
+    let level = match convert_cstr("log_level", log_level) {
```

Comment:
You can use `unwrap_or_default` in this call and the one beneath instead of doing the pattern match. There should be a lint for that I think, but dunno if this crate has customised the lint rules?